modules/tinymce/src/plugins/table/main/ts/api/Commands.ts

```diff
@@ -77,6 +166,47 @@ const registerCommands = (editor: Editor, actions: TableActions, cellSelection:
     Events.fireTableModified(editor, table.dom, data.effect);
   };
 
+  const toggleTableClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCellOrCaption(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        if (Class.has(table, requestedClass)) {
+          Class.remove(table, requestedClass);
+        } else {
+          Class.add(table, requestedClass);
+        }
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
+      });
+    });
+  };
+
+  const toggleTableCellClass = (_ui: boolean, requestedClass: string) => {
+    getSelectionStartCell(editor).each((startCell) => {
+      TableLookup.table(startCell, isRoot).filter(Fun.not(isRoot)).each((table) => {
+        const cells = TableSelection.getCellsFromSelection(startCell, selections);
+
+        const warehouse = Warehouse.fromTable(table);
+        const allCells = Warehouse.justCells(warehouse);
+
+        const filtered = Arr.filter(allCells, (cellA) =>
+          Arr.exists(cells, (cellB) =>
+            Compare.eq(cellA.element, cellB)
+          )
+        );
+
+        Arr.each(filtered, (value) => {
+          if (Class.has(value.element, requestedClass)) {
+            Class.remove(value.element, requestedClass);
+          } else {
+            Class.add(value.element, requestedClass);
+          }
+        });
+
+        Events.fireTableModified(editor, table.dom, Events.structureModified);
```

Comment:
This is definitely the wrong thing to be firing here, as this isn't changing the structure and instead is only changing the styling.

Reply:
The event is correct, it's the data that's wrong. It should be `Events.styleModified` instead.